src/datarobotx/idp/custom_application_source.py

```diff
@@ -0,0 +1,316 @@
+#
```

Comment:
on the surface - feels like this should be two files to maintain precedent/consistency, e.g. custom_application_source.py, custom_application_source_version.py; i also see this is how you organized your tests